src/main/java/shukaro/warptheory/handlers/WarpHandler.java

```diff
@@ -181,6 +184,22 @@ public static void purgeWarp(EntityPlayer player)
         removeWarp(player, getTotalWarp(player));
     }
 
+	//add function to remove 1 warp, chance based
+	//warp < 50 = 0%, warp >= 150 = 100%
+    public static void purgeWarpMinor(EntityPlayer player)
+    {
+		Random ran = new Random(); 
```

Comment:
No point allocating a new Random instance every single time..
Add it as a class member `private static final chanceRandom = new Random();`